sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -231,6 +260,8 @@ static az_result connecting(az_event_policy* me, az_event event)
   az_result ret = AZ_OK;
   az_mqtt5_connection* this_policy = (az_mqtt5_connection*)me;
 
+  // Start and stop the timer on entering and exiting.
```

Comment:
Nit 1: comment is far from the actual code.
Nit 2: there is no timer (such as in ReconnectTimeout), just a time-interval measurement of how long we stayed in the Connecting state.